content/blog/2025-04-10-fastest-tpch-generator.md

```diff
@@ -0,0 +1,614 @@
+263|2396|649|1|22|28564.58|0.06|0.08|R|F|1994-08-24|1994-06-20|1994-09-09|NONE|FOB|efully express fo|
+327|4172|427|2|9|9685.53|0.09|0.05|A|F|1995-05-24|1995-07-11|1995-06-05|NONE|AIR| asymptotes are fu|
+450|5627|393|4|40|61304.80|0.05|0.03|R|F|1995-03-20|1995-05-25|1995-04-14|NONE|RAIL|ve. asymptote|
+```
+
+**Figure 2**: Example TBL formatted output of `dbgen` for the `LINEITEM` table
+
+<table>
+  <tr>
+   <td><strong>Scale Factor</strong>
+   </td>
+   <td><strong>Data Size (TBL)</strong>
+   </td>
+   <td><strong>Data Size (Parquet)</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>0.1
+   </td>
+   <td>103 Mb
+   </td>
+   <td>31 Mb
+   </td>
+  </tr>
+  <tr>
+   <td>1
+   </td>
+   <td>1 Gb
+   </td>
+   <td>340 Mb
+   </td>
+  </tr>
+  <tr>
+   <td>10
+   </td>
+   <td>10 Gb
+   </td>
+   <td>3.6 Gb
+   </td>
+  </tr>
+  <tr>
+   <td>100
+   </td>
+   <td>107 Gb
+   </td>
+   <td>38 Gb
+   </td>
+  </tr>
+  <tr>
+   <td>1000
+   </td>
+   <td>1089 Gb
+   </td>
+   <td>379 Gb
+   </td>
+  </tr>
+</table>
+
+
+**Table 1**: TPC-H data set sizes at different scale factors for both TBL and [Apache Parquet].
+
+[Apache Parquet]: https://parquet.apache.org/
+
+# Why do we need a new TPC-H Data generator?
+
+Despite the known limitations of the TPC-H benchmark, it is so well known that it
+is used frequently in database performance analysis. To run TPC-H, you must first
+load the data, using `dbgen`, which is not ideal for several reasons:
+
+1. You must find and compile a copy of the 15+ year old C program (for example [electrum/tpch-dbgen])
+2. `dbgen` requires substantial time (Figure 3) and is not able to use more than one core.
+3. It outputs TBL format, which typically requires loading into your database (for example, [here is how to do so] in Apache DataFusion) prior to query.
+4. The implementation makes substantial assumptions about the operating environment, making it difficult to extend or embed into other systems.<sup>2</sup>
+
+[electrum/tpch-dbgen]: https://github.com/electrum/tpch-dbgen
+[here is how to do so]: https://github.com/apache/datafusion/blob/507f6b6773deac69dd9d90dbe60831f5ea5abed1/datafusion/sqllogictest/test_files/tpch/create_tables.slt.part#L24-L124
+ 
+
+<img src="/blog/images/fastest-tpch-generator/tbl-performance.png" alt="Time to generate TPC-H data in TBL format" width="80%" class="img-responsive">
+
+**Figure 3**: Time to generate TPC-H data in TBL format. The default `tpchgen` is
+shown in blue. `tpchgen` restricted to a single core is shown in red. Unmodified
+`dbgen` is shown in green and `dbgen` modified to use `-O3` optimization level
+is shown in yellow.
+
+`dbgen` is so inconvenient and takes so long that vendors often provide
+preloaded TPC-H data, for example [Snowflake Sample Data], [DataBricks Sample
+datasets] and [DuckDB Pre-Generated Data Sets].
+
+[Snowflake Sample  Data]: https://docs.snowflake.com/en/user-guide/sample-data-tpch
```

Comment:
hyperlink not generated 

![Screenshot 2025-04-05 at 11 12 38 AM](https://github.com/user-attachments/assets/004cd640-9c44-4682-83cb-af1296d0065e)

```suggestion
[Snowflake Sample Data]: https://docs.snowflake.com/en/user-guide/sample-data-tpch
```